server/sonar-process-monitor/src/main/java/org/sonar/application/SchedulerImpl.java

```diff
@@ -0,0 +1,278 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.application;
+
+import java.util.EnumMap;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.application.config.AppSettings;
+import org.sonar.application.config.AppSettingsHelper;
+import org.sonar.application.process.JavaCommand;
+import org.sonar.application.process.JavaCommandFactory;
+import org.sonar.application.process.JavaProcessLauncher;
+import org.sonar.application.process.Lifecycle;
+import org.sonar.application.process.ProcessEventListener;
+import org.sonar.application.process.ProcessLifecycleListener;
+import org.sonar.application.process.SQProcess;
+import org.sonar.process.ProcessId;
+
+public class SchedulerImpl implements Scheduler, ProcessEventListener, ProcessLifecycleListener, AppStateListener {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SchedulerImpl.class);
+
+  private final AppSettings settings;
+  private final AppReloader appReloader;
+  private final JavaCommandFactory javaCommandFactory;
+  private final JavaProcessLauncher javaProcessLauncher;
+  private final AppState appState;
+  private final NodeLifecycle nodeLifecycle = new NodeLifecycle();
+
+  private final CountDownLatch keepAlive = new CountDownLatch(1);
+  private final AtomicBoolean restartRequested = new AtomicBoolean(false);
+  private final AtomicBoolean restartDisabled = new AtomicBoolean(false);
+  private final EnumMap<ProcessId, SQProcess> processesById = new EnumMap<>(ProcessId.class);
+  private final AtomicInteger operationalCountDown = new AtomicInteger();
+  private final AtomicInteger stopCountDown = new AtomicInteger();
+  private StopperThread stopperThread;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove the "stopperThread" field and declare it as a local variable in the relevant methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1450)